### ISSUE TITLE
[Feature] Add deterministic training mode in Keras

### ISSUE THREAD

**Issue by amadhan882** [keras-team/keras#22621]
## Description

Currently, achieving reproducible training results in Keras is non-trivial, even when explicitly setting random seeds. In practice, identical seeds, data, and model configurations can still produce significantly different outputs.

There is no unified, high-level way in Keras to enforce deterministic behavior across:

- layer initialization  
- optimizer updates  
- backend operations  

---

## Rationale

Reproducibility is essential for:

- scientific experiments  
- debugging workflows  
- model comparison and validation  
- production reliability  

At present, users must manually configure backend-specific settings, and deterministic behavior is not consistently guaranteed.

---

## Proposed Feature

Introduce a simple, high-level API:

```python
keras.config.enable_determinism()
```

This would:

- enforce deterministic backend behavior where possible  
- configure random seeds consistently across components  
- disable or warn about nondeterministic operations  
- provide a unified and user-friendly reproducibility mode  

---

## Proof of Concept

The following example demonstrates non-deterministic behavior despite fixed seeds:

```python
import os
os.environ["CUDA_VISIBLE_DEVICES"] = ""  # Force CPU

import keras
from keras import layers
import numpy as np
import tensorflow as tf

# Fix seeds
np.random.seed(42)
tf.random.set_seed(42)

def build_model():
    model = keras.Sequential([
        layers.Dense(16, activation="relu"),
        layers.Dense(4)
    ])
    model.compile(optimizer="adam", loss="mse")
    return model

# Fixed data
x = np.random.randn(100, 8).astype("float32")
y = np.random.randn(100, 4).astype("float32")

# Run 1
model1 = build_model()
model1.fit(x, y, epochs=3, verbose=0)
out1 = model1(x)

# Reset seeds
np.random.seed(42)
tf.random.set_seed(42)

# Run 2
model2 = build_model()
model2.fit(x, y, epochs=3, verbose=0)
out2 = model2(x)

# Compare outputs
diff = np.max(np.abs(out1 - out2))
print("Max difference:", diff)
```

---

## Observed Behavior

```
Max difference: 2.7+
```

---

## Expected Behavior

With deterministic mode enabled:

```python
keras.config.enable_determinism()
```

Expected output:

```
Max difference: ~0 (or near machine precision)
```

---

## Impact

- Improves reproducibility across training runs  
- Simplifies debugging and experimentation  
- Aligns with expectations in research and production environments  
- Reduces hidden nondeterministic behavior  

---

## Additional Notes

TensorFlow provides partial support for deterministic execution, but this is not exposed through a simple, unified Keras API.

A Keras-level abstraction would significantly improve usability and reliability.

**Comment by hertschuh** [comment]
This already exists. It's `keras.utils.set_random_seed`.

Also note that you had your "Fix seed" at wrong place before.

```python
import os
os.environ["CUDA_VISIBLE_DEVICES"] = ""  # Force CPU

import keras
from keras import layers
import numpy as np
import tensorflow as tf

def build_model():
    model = keras.Sequential([
        layers.Dense(16, activation="relu"),
        layers.Dense(4)
    ])
    model.compile(optimizer="adam", loss="mse")
    return model

# Fixed data
x = np.random.randn(100, 8).astype("float32")
y = np.random.randn(100, 4).astype("float32")

# Fix seeds
keras.utils.set_random_seed(42)

# Run 1
model1 = build_model()
model1.fit(x, y, epochs=3, verbose=0)
out1 = model1(x)

# Reset seeds
keras.utils.set_random_seed(42)

# Run 2
model2 = build_model()
model2.fit(x, y, epochs=3, verbose=0)
out2 = model2(x)

# Compare outputs
diff = np.max(np.abs(out1 - out2))
print("Max difference:", diff)
```

```
Max difference: 0.0
```
